src/test/fuzz/util.h

```diff
@@ -512,4 +512,63 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSocket : public MockableSocket
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSocket(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ssize_t send(const char* buf, size_t len, int flags) override
+    {
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            return len;
+        }
+        return m_fuzzed_data_provider.ConsumeIntegralInRange<ssize_t>(-1, len);
```

Comment:
I think it would be good to set `errno` if we are going to return an error (`-1`). Application reaction on error should differ depending on the value of `errno` (should retry on "temporary" errors).

Reply:
Good point! Pushed an updated version addressing this.